App.tsx

```diff
@@ -5,15 +5,33 @@ import { useFont } from './shared/hooks/useFont';
 import { GlobalContextProvider } from './components/GlobalContextProvider';
 import { GlobalContext } from './shared/GlobalContext';
 import { useSelector } from '@xstate/react';
-import { selectIsReady } from './machines/app';
-
-import './i18n';
+import { useTranslation } from 'react-i18next';
+import { selectIsReadError, selectIsReady } from './machines/app';
+import { MessageOverlay } from './components/MessageOverlay';
+import { useApp } from './screens/AppController';
 
 const AppInitialization: React.FC = () => {
   const { appService } = useContext(GlobalContext);
   const hasFontsLoaded = useFont();
   const isReady = useSelector(appService, selectIsReady);
-
+  const isReadError = useSelector(appService, selectIsReadError);
+  const controller = useApp();
+  const { t } = useTranslation('WelcomeScreen');
+  if (isReadError) {
+    return (
+      <MessageOverlay
+        isVisible={isReadError}
+        title={t('failedToReadKeys')}
+        message={t('retryRead')}
+        onTryAgain={() => {
+          controller.TRY_AGAIN();
```

Comment:
Direct assignment can be done here as both does the same job -> onTryAgain={TRY_AGAIN}